tests/integration/test_vulnerability_detector/test_providers/test_providers_no_os.py

```diff
@@ -57,115 +57,154 @@
+                 if errors occur when the <os> tag is omitted in the configuration in providers that should have it and,
+                 on the other hand, if the update of the feeds starts normally in providers that do not require this
+                 tag.
+
+    wazuh_min_version: 4.2.0
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Wazuh configuration data. Needed for set_wazuh_configuration fixture.
+        - metadata:
+            type: dict
+            brief: Wazuh configuration metadata
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set the wazuh configuration according to the configuration data.
+        - truncate_log_files:
+            type: fixture
+            brief: Truncate the log files at the end of the testing case.
+        - clean_cve_tables_func:
+            type: fixture
+            brief: Clean all the CVE tables before and after running the test.
+        - stop_modules_function:
+            type: fixture
+            brief: Stop the wazuh modules daemon.
 
-metadata = [
-    {'provider_name': 'Amazon Linux', 'error': 1},
-    {'provider_name': 'Ubuntu', 'error': 1},
-    {'provider_name': 'Debian', 'error': 1},
-    {'provider_name': 'Red Hat Enterprise Linux', 'os': ['5', '6', '7', '8', '']},
-    {'provider_name': 'National Vulnerability Database', 'os': ['']},
-    {'provider_name': 'Microsoft Security Update', 'os': ['']},
-    {'provider_name': 'Arch Linux', 'os': ['']},
-]
+    assertions:
+        - The provider os data update starts when `os` has not a determined value.
+        - The error message appears when `os` tag is not present in Amazon Linux, Canonical and Debian providers.
 
-ids = [f"{item['PROVIDER']}" for item in params]
+    input_description:
+        - The `test_no_os.yaml` file provides the module configuration for this test.
 
-# configuration data
-configurations = load_wazuh_configurations(configurations_path, __name__, params=params, metadata=metadata)
+    expected_output:
+        - r'.*: Configuration error at.*'
+        - 'Starting <provider_name>  database update'
+    '''
+    provider_name = metadata['provider_name']
 
+    # Those providers that aren't expected to work without the <os> tag.
+    try:
+        control_service('restart')
+    except ValueError:
+        evm.check_configuration_error()
+    else:
+        os = metadata['os'][0]
+        if os != '':
+            os_name = f"{provider_name} {os}"
+        else:
+            os_name = f"JSON {provider_name}" if 'Red Hat' in provider_name else f"{provider_name}"
 
-# fixtures
-@pytest.fixture(scope="module", params=configurations, ids=ids)
-def get_configuration(request):
-    """Get configurations from the module."""
-    return request.param
+        evm.check_provider_database_update_start_log(os_name, vd.DOWNLOAD_TIMEOUT)
```

Comment:
It is necessary this timeout `vd.DOWNLOAD_TIMEOUT` here?

Reply:
If I leave the default timeout It will fail with the Timeout message.